tensorflow/core/util/tensor_bundle/byte_swap_tensor.cc

```diff
@@ -161,6 +161,11 @@ absl::Status ByteSwapTensor(Tensor* t) {
                         t->NumElements());
 }
 
+absl::Status ByteSwapTensorProto(TensorProto* tp) {
+  char* buff = const_cast<char*>((tp->tensor_content().data()));
```

Comment:
```suggestion
  char* buff = const_cast<char*>(std::string((tp->tensor_content()).data()));
```

I think we need to wrap this into a `std::string` constructor here, see https://github.com/tensorflow/tensorflow/blob/b77b79bcee488454292188cce47661ab854fe401/tensorflow/core/util/tensor_bundle/byte_swap_tensor.cc#L187

Without this, we have several compile failures internally as the `tensor_content` output is not always a type that supports `.data()`.

Reply:
(repeating suggestion so I could apply it)

```suggestion
  char* buff = const_cast<char*>(std::string((tp->tensor_content()).data()));
```

Reply:
Thanks @mihaimaruseac . Let me build and retest here and then I'll push the change.

Reply:
@mihaimaruseac , I had to set the byte swapped TensorProto content back into the TensorProto since the buffer was getting a copy of the content but otherwise no other changes to your suggested change. I have tested it on amd64 and s390x linux.

Reply:
Thank you! I missed the need to set the content back into the proto, but it makes sense.